03.bowling/bowling.rb

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+def make_frames(shots)
+  frames = []
+  shots.each_slice(2) do |s|
+    frames << if s[0] == 10
+                [s[0]]
+              else
+                s
+              end
+  end
+  frames
+end
+
+def cal_point(frames)
+  point = 0
+  frames.each_with_index do |frame, index|
+    break if index >= 10
+
+    point += if frame[0] == 10
+               frame.sum + cal_extra_point(frames[index + 1..], 2)
+             elsif frame.sum == 10
+               frame.sum + cal_extra_point(frames[index + 1..], 1)
+             else
+               frame.sum
```

Comment:
```suggestion
               frame.sum + cal_extra_point(frames[index + 1..], 0)
```

というふうに考えるとそれぞれ違うのは引数の2,1,0だけなので、ロジックをもっとDRYにできそうです